apps/storefront/src/pages/Invoice/index.test.tsx

```diff
@@ -1350,3 +1366,176 @@ it('exports selected invoices as CSV', async () => {
     expect(window.open).toHaveBeenCalledWith('https://example.com/invoices.csv', '_blank');
   });
 });
+
+describe('when the url contains an invoiceId parameter', () => {
+  it('display invoices matching the invoiceId and expands them', async () => {
+    const pdfFile = new Blob(['%PDF-1.4 Mock PDF Content'], { type: 'application/pdf' });
+
+    const getInvoicesResponse = vi.fn();
+    const getInvoicePDFUrlResponse = vi.fn();
+
+    server.use(
+      graphql.query('GetInvoices', ({ query }) => HttpResponse.json(getInvoicesResponse(query))),
+      graphql.query('GetInvoiceStats', () =>
+        HttpResponse.json(buildInvoiceStatsResponseWith('WHATEVER_VALUES')),
+      ),
+      graphql.mutation('GetInvoicePDFUrl', ({ query }) =>
+        HttpResponse.json(getInvoicePDFUrlResponse(query)),
+      ),
+      http.get('https://example.com/123456.pdf', async () =>
+        HttpResponse.arrayBuffer(await pdfFile.arrayBuffer(), {
+          headers: { 'Content-Type': 'application/pdf' },
+        }),
+      ),
+      http.get('https://example.com/123478.pdf', async () =>
+        HttpResponse.arrayBuffer(await pdfFile.arrayBuffer(), {
+          headers: { 'Content-Type': 'application/pdf' },
+        }),
+      ),
+    );
+
+    when(getInvoicePDFUrlResponse)
+      .calledWith(stringContainingAll('123456'))
+      .thenReturn({ data: { invoicePdf: { url: 'https://example.com/123456.pdf' } } });
+
+    when(getInvoicePDFUrlResponse)
+      .calledWith(stringContainingAll('123478'))
+      .thenReturn({ data: { invoicePdf: { url: 'https://example.com/123478.pdf' } } });
+
+    when(getInvoicesResponse)
+      .calledWith(stringContainingAll('search: "1234"'))
+      .thenReturn(
+        buildInvoicesResponseWith({
+          data: {
+            invoices: {
+              edges: [
+                buildInvoiceWith({ node: { id: '123456' } }),
+                buildInvoiceWith({ node: { id: '123478' } }),
+              ],
+            },
+          },
+        }),
+      );
+
+    renderWithProviders(<Invoice />, {
+      preloadedState,
+      initialEntries: [{ search: '?invoiceId=1234' }],
+    });
+
+    const { log } = console;
+    vi.spyOn(console, 'log').mockImplementation(log);
+
+    // PDFObject will log a warning in the console if it cannot embed the PDF
+    // eslint-disable-next-line no-console
+    when(console.log)
```

Comment:
we see the logging as a feature that we want to protect?

(i.e. `when` vs just an unconditional, noop overwrite)